templates/decays.hpp.in

```diff
@@ -0,0 +1,236 @@
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+
+/**
+ * @file @ModelName@_decays.hpp
+ *
+ * @brief contains class for calculating particle decays
+ *
+ * This file was generated at @DateAndTime@ with FlexibleSUSY
+ * @FlexibleSUSYVersion@ (git commit: @FlexibleSUSYGitCommit@) and SARAH @SARAHVersion@ .
+ */
+
+#ifndef @ModelName@_DECAYS_H
+#define @ModelName@_DECAYS_H
+
+#include "@ModelName@_decay_table.hpp"
+#include "@ModelName@_mass_eigenstates.hpp"
+#include "@ModelName@_mass_eigenstates_decoupling_scheme.hpp"
+#include "@ModelName@_input_parameters.hpp"
+#include "cxx_qft/@ModelName@_qft.hpp"
+#include "@ModelName@_decay_amplitudes.hpp"
+#include "decays_loop_corrections.hpp"
+#include "decays_problems.hpp"
+#include "lowe.h"
+#include "wrappers.hpp"
+#include "error.hpp"
+
+namespace flexiblesusy {
+
+template <typename Field1, typename Field2>
+double
+final_state_symmetry_factor(typename @ModelName@_cxx_diagrams::field_indices<Field1>::type const& idx1,
+                            typename @ModelName@_cxx_diagrams::field_indices<Field2>::type const& idx2)
+{
+   if (std::is_same<Field1, Field2>::value) {
+      // TODO: we also compare if (potential) color indices are the same,
+      //       this might cause problems if someone accidentaly initializes them
+      //       differently
+      if (boost::range::equal(idx1, idx2)) {
+         return 0.5;
+      }
+   }
+   return 1.;
+}
+
+enum class HigherOrderSMCorrections {enable, disable};
+
+
+class @ModelName@_decays {
+public:
+   @ModelName@_decays() = default;
+   @ModelName@_decays(@ModelName@_mass_eigenstates model_, softsusy::QedQcd const& qedqcd_, @ModelName@_input_parameters const& input_,
+         HigherOrderSMCorrections enable_corrections = HigherOrderSMCorrections::enable)
+      : model(model_)
+      , input(input_)
+      , qedqcd(qedqcd_)
+      , include_higher_order_corrections(enable_corrections)
+      {}
+   @ModelName@_decays(const @ModelName@_decays&) = default;
+   @ModelName@_decays(@ModelName@_decays&&) = default;
+   ~@ModelName@_decays() = default;
+   @ModelName@_decays& operator=(const @ModelName@_decays&) = default;
+   @ModelName@_decays& operator=(@ModelName@_decays&&) = default;
+
+   void set_loop_corrections(const Decays_loop_corrections&);
+   const Decays_loop_corrections& get_loop_corrections() const;
+   const @ModelName@_decay_table& get_decay_table() const;
+   const Decays_problems& get_problems() const;
+
+   void clear();
+   void clear_problems();
+   void calculate_decays();
+
+@decaysGetters@
+@decaysCalculationPrototypes@
+
+@partialWidthCalculationPrototypes@
+
+private:
+   @ModelName@_mass_eigenstates model{};
+   @ModelName@_input_parameters input{};
```

Comment:
This field is redundant, because the `model` object already contains the input parameters. One can extract them by calling `model.get_input()`.

Reply:
Done in 750d4ba6b4260f7372ae2d726179c810ca0c00ab.

Reply:
Thanks a lot! :)